app/src/main/java/com/github/se/eventradar/model/event/EventDetailsViewModel.kt

```diff
@@ -13,76 +15,98 @@ import kotlinx.coroutines.flow.StateFlow
-          } else {
-            // Document does not exist
-            println("Document does not exist")
-          }
-        }
-        .addOnFailureListener { exception ->
-          // Handle any errors that occur during fetching
-          // You may want to log the error or show a message to the user
-          println("Error fetching event: $exception")
-        }
+  private fun createEventFromDocument(document: DocumentSnapshot, eventId: String): Event {
+    val startInstant: Instant = document.getTimestamp("start")!!.toDate().toInstant()
+    val endInstant: Instant = document.getTimestamp("end")!!.toDate().toInstant()
+    val latitude = document.getGeoPoint("location")!!.latitude
+    val longitude = document.getGeoPoint("location")!!.longitude
+    val address = document.getString("address") ?: ""
+    val ticketName = document.getString("ticketName") ?: ""
+    val ticketPrice = document.getLong("ticketPrice")!!.toDouble()
+    val ticketQuantity = document.getLong("ticketQuantity")!!.toInt()
+    val category = document.getString("category") ?: ""
+
+    return Event(
+        eventName = document.getString("name") ?: "",
+        eventPhoto = document.getString("photo") ?: "",
+        start = LocalDateTime.ofInstant(startInstant, ZoneId.systemDefault()),
+        end = LocalDateTime.ofInstant(endInstant, ZoneId.systemDefault()),
+        location = Location(latitude, longitude, address),
+        description = document.getString("description") ?: "",
+        ticket = Ticket(ticketName, ticketPrice, ticketQuantity),
+        contact = document.getString("contact") ?: "",
+        organiserList = (document["organiserList"] as List<*>).map { it.toString() }.toSet(),
+        attendeeList = (document["attendeeList"] as List<*>).map { it.toString() }.toSet(),
+        category = getCategory(category),
+        fireBaseID = eventId,
+    )
   }
-}
 
-private fun initialEventState(): EventUiState {
-  return EventUiState(
-      "",
-      "",
-      LocalDateTime.MIN,
-      LocalDateTime.MAX,
-      Location(0.0, 0.0, ""),
-      "",
-      Ticket("", 0.0, 0),
-      "",
-      EventCategory.MUSIC,
-  )
+  private fun getCategory(value: String): EventCategory {
+    var category = EventCategory.UNDEFINED
+    try {
+      category = EventCategory.valueOf(value)
+    } catch (e: IllegalArgumentException) {
+      Log.d(tag, "No valid category")
+    }
+    return category
+  }
 }
 
 data class EventUiState(
-    var eventName: String,
-    var eventPhoto: String,
-    var start: LocalDateTime,
-    var end: LocalDateTime,
-    var location: Location,
-    var description: String,
-    var ticket: Ticket,
-    var contact: String,
-    var category: EventCategory,
+    val eventName: String = "",
+    val eventPhoto: String = "",
+    val start: LocalDateTime = LocalDateTime.MIN,
+    val end: LocalDateTime = LocalDateTime.MAX,
+    val location: Location = Location(0.0, 0.0, ""), // TODO add standard val to location class ?
```

Comment:
Sorry, I should have read my comments before pushing. I meant a default value. One that we can use by default if there's problem with the database data (similar to the UNDEFINED EventCategory that I added in case the "category field" doesn't contain a valide category)